### PR TITLE
refactor: use resource watches on dashboard

### PR DESCRIPTION
**Pull request by utkuozdemir**
Instead of doing excessive get/list requests, do a watch per node in an infinite retry.

Additionally, refactor the dashboard code to make the various data listener namings more consistent and reorganize the packages.

Closes siderolabs/talos#6960.

### CODE REVIEW

File: internal/pkg/dashboard/apidata/source.go [hunk 1]
```diff
@@ -85,23 +82,36 @@ func (source *API) run(dataCh chan<- *data.Data) {
 }
 
 //nolint:gocyclo,cyclop
-func (source *API) gather() *data.Data {
-	result := &data.Data{
+func (source *Source) gather() *Data {
```
Comment (utkuozdemir) [inline internal/pkg/dashboard/apidata/source.go:85]: Reverted API data gathering logic here to its previous shape, as we now don't need to do anymore an initial request to find out the set of nodes.

File: internal/pkg/dashboard/components/footer.go [hunk 2]
```diff
@@ -19,13 +18,12 @@ import (
 type Footer struct {
 	tview.TextView
 
-	nodes             []string
-	selectedNodeIndex int
+	selectedNode string
```
Comment (utkuozdemir) [inline internal/pkg/dashboard/components/footer.go:21]: Moved the node selection logic from the footer to the dashboard.go itself & made it simpler.

File: internal/pkg/dashboard/dashboard.go [hunk 3]
```diff
@@ -46,14 +48,24 @@ const (
...
+
+// LogDataListener is a listener which is notified when a log line is received.
+type LogDataListener interface {
+	OnLogDataChange(node string, logLine string)
+}
+
+// NodeSetListener is a listener which is notified when the set of nodes changes.
+type NodeSetListener interface {
+	OnNodeSetChange(nodes []string)
 }
```
Comment (utkuozdemir) [inline internal/pkg/dashboard/dashboard.go:69]: The idea here is to make the 3 different data sources and their listeners namings similar to each other.

File: internal/integration/api/apply-config.go [hunk 4]
```diff
@@ -239,22 +239,18 @@ func (suite *ApplyConfigSuite) TestApplyConfigRotateEncryptionSecrets() {
 
 	encryption := machineConfig.MachineConfig.MachineSystemDiskEncryption
 
-	if encryption == nil {
+	if encryption == nil { //nolint:staticcheck
 		suite.T().Skip("skipped in not encrypted mode")
 	}
 
-	cfg := encryption.EphemeralPartition
```
Comment (smira) [inline internal/integration/api/apply-config.go:246]: this looks weird, could it be that your `make lint` got wrong cache and now reporting wrong issues?

Comment (utkuozdemir) [reply internal/integration/api/apply-config.go:246]: oh yes, must be that, how can I fix it?

File: internal/pkg/dashboard/resourcedata/resourcedata.go [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+
+			return
+		case <-timer.C:
+		}
+	}
+}
+
+//nolint:gocyclo,cyclop
+func (source *Source) runResourceWatch(ctx context.Context, node string) error {
+	if node != "" {
```
[the remark this answers is not in the document]
Comment (utkuozdemir) [reply internal/pkg/dashboard/resourcedata/resourcedata.go:110]: I do it here? https://github.com/siderolabs/talos/pull/6967/files#diff-5a885872d568c5cd54266f334a33749652c59927321d5443c914dc469b9c014cR85-R89

[the remark this answers is not in the document]
Comment (smira) [reply internal/pkg/dashboard/resourcedata/resourcedata.go:110]: I would better move it into the function with `defer`, easier to understand the flow and handles panics as well

File: internal/integration/api/apply-config.go (hunk 4 above)
Comment (smira) [reply internal/integration/api/apply-config.go:246]: sometimes just `docket stop` for the buildx container is enough

sometimes `docker rm -f` for the buildx container helps (as it kills the cache)

you can also try `docker buildx prune`

**Review by smira — APPROVED** [review]

File: internal/integration/api/apply-config.go (hunk 4 above)
Comment (utkuozdemir) [reply internal/integration/api/apply-config.go]: yep, the last command solved it. reverted.

File: internal/pkg/dashboard/resourcedata/resourcedata.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (utkuozdemir) [reply internal/pkg/dashboard/resourcedata/resourcedata.go:104]: done

**Pull request merged**